state/podspec.go

```diff
@@ -43,16 +44,21 @@ func (m *CAASModel) SetPodSpec(appTag names.ApplicationTag, spec string) error {
 			C:  podSpecsC,
 			Id: applicationGlobalKey(appTag.Id()),
 		}
-		existing, err := m.PodSpec(appTag)
+		existing, err := m.podInfo(appTag)
 		if err == nil {
-			if existing == spec {
-				return nil, jujutxn.ErrNoOperations
+			updates := bson.D{{"$inc", bson.D{{"upgrade-counter", 1}}}}
+			if spec != nil {
+				updates = append(updates, bson.DocElem{"$set", bson.D{{"spec", *spec}}})
```

Comment:
No, because that would break what we're trying to achieve. Passing in nil doesn't mean to set the spec to "", it means increment the counter without changing the spec. If we reset the spec to "" we would wipe out what the charm has previously written.